front/src/modules/trainschedule/components/ImportTrainSchedule/ImportTrainScheduleConfig.tsx

```diff
@@ -170,28 +170,22 @@ const ImportTrainScheduleConfig = ({
     startDate: string
   ): Step[] =>
     ocpTTs
-      .map((ocpTT, index): Step | null => {
+      .map((ocpTT): Step | null => {
         const ocpRef = ocpTT.getAttribute('ocpRef');
         const times = ocpTT.getElementsByTagName('times')[0];
+        const isLastOcp = ocpTT === ocpTTs.at(-1);
+        const ocpType = ocpTT.getAttribute('ocpType');
         let departureTime = times?.getAttribute('departure') || '';
-        let arrivalTime = times?.getAttribute('arrival') || '';
-
-        const isLastOcpTT = index === ocpTTs.length - 1;
-
-        if (isLastOcpTT) {
-          arrivalTime = cleanTimeFormat(departureTime) || cleanTimeFormat(arrivalTime); // For the last sequence, arrival equals departure
-          departureTime = cleanTimeFormat(arrivalTime) || cleanTimeFormat(departureTime);
-        } else if (index !== 0) {
-          arrivalTime = times?.getAttribute('arrival') || times?.getAttribute('departure') || '';
-          arrivalTime = cleanTimeFormat(arrivalTime);
-          departureTime = times?.getAttribute('departure') || times?.getAttribute('arrival') || '';
-          departureTime = cleanTimeFormat(departureTime);
-        }
+        let arrivalTime =
+          ocpType === 'pass' ? departureTime || '' : times?.getAttribute('arrival') || '';
```

Comment:
Nit: `departureTime || ''` can be simplified to `departureTime`, because the line right above already has `|| ''`.

Reply:
thanks 🙏